common/types/activation.go

```diff
@@ -191,11 +132,12 @@ type InnerActivationTx struct {
 
 	NIPost      *NIPost
 	InitialPost *Post
+	NodeID      *NodeID
 	VRFNonce    *VRFPostIndex
 
 	// the following fields are kept private and from being serialized
-	id                *ATXID    // non-exported cache of the ATXID
-	nodeID            *NodeID   // the id of the Node that created the ATX (public key)
+	id                ATXID     // non-exported cache of the ATXID
+	nodeID            NodeID    // non-exported cache of the NodeID that created the ATX
```

Comment:
why do we still need `nodeID` when `ActivationTx.SmesherID` is always set?

Reply:
Updated to remove `nodeId`